### ISSUE TITLE
Cross references break with special characters

### ISSUE THREAD

**Issue by nitrocalcite** [asciidoctor/asciidoctor#3642]
This is a pretty minor issue, but I don't think it's intended.

If special characters like "..." are used in a section title, cross references to that section are displayed incorrectly.  Here's a MWE:

```
= Title
nitrocalcite

== How do I...
Some body text

== Another section
I want to reference the prior section <<How do I...>>
```
Both the text and link are broken.
An obvious workaround is to give the section with the offending title an explicit ID and reference that instead.  This in fact works, and is generally better regardless.

Thanks for this awesome library!  I had been wondering why we couldn't have Markdown syntax at LaTeX scale, until I found this.

**Comment by mojavelinux** [comment]
Thanks for the high praise! And welcome!

Although I do generally recommend using an ID, esp in this case, I'm kind of surprised this particular case breaks. The order of substitutions is right, so in theory it should work. I'll investigate.

Keep in mind that the `<<Title>>` shorthand is meant for rapid prototyping...as it's generally less stable over the long term than using IDs (and explicit IDs give you the greatest stability).

**Comment by mojavelinux** [comment]
I see what's happening. The `#` in the character reference is getting interpreted as the delimiter for a URL fragment on the target. Here's what the parser sees:

```
Hello there&#8230;&#8203;
```

A very simple fix is to ensure that the preceding character is not `&`, as that almost certainly indicates it's a character reference.

**Comment by mojavelinux** [comment]
This ended up getting resolved by #4393.

**Comment by mojavelinux** [comment]
> A very simple fix is to ensure that the preceding character is not &, as that almost certainly indicates it's a character reference.

This is precisely what we ended up doing to solve this.
